images/filelogoffsetsynch/src/filelogoffsetsynch.go

```diff
@@ -0,0 +1,336 @@
+// SPDX-FileCopyrightText: Copyright 2024 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type settings struct {
+	Clientset                  *kubernetes.Clientset
+	NodeName                   string
+	ConfigMapNamespace         string
+	ConfigMapName              string
+	FilelogOffsetDirectoryPath string
+}
+
+type patch struct {
+	BinaryData map[string]string `json:"binaryData,omitempty"`
+}
+
+// TODO Add support for sending_queue on separate exporter
+// TODO Set up compaction
+// TODO Set up metrics & logs
```

Comment:
@mmanciop a couple of TODOs left in the code here, are they meant to stay? Delete? Convert to Linear issues?